internal/storage/db.go

```diff
@@ -0,0 +1,274 @@
+package storage
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/placer14/gof-server/internal/database"
+	"github.com/placer14/gof-server/internal/handlers/payloads"
+	"gorm.io/datatypes"
+)
+
+type key_db int
+
+const KeyDBStorage key_db = iota
+
+type DBStorage struct {
+}
+
+var _ Storageable = &DBStorage{}
+
+func (s *DBStorage) GetBool(key string) (bool, error) { return GetFlag[bool](key) }
```

Comment:
I'm not sure how well it would work, but maybe generics could simplify this code a bit? For example, the functions could abstract over a private generic function. The benefit would be one piece of code handling all of the types and less surface to maintain or where you could add hooks (telemetry measurements, storage middleware, etc)